src/app/core/actions/transform/TranslateAction.ts

```diff
@@ -1,40 +1,63 @@
 import {Vector} from "Vector";
-
 import {Action} from "core/actions/Action";
-
 import {Component} from "core/models/Component";
-
 import {SnapPos} from "./SnapUtils";
 
-// Translate can be applied to single components,
-//  but if you need to translate multiple components at
-//  once you MUST do it as a group to avoid the issue #417
-//  https://github.com/OpenCircuits/OpenCircuits/issues/417
+/**
+ * Translate can be applied to single components or groups of components,
+ * used for moving componets from one position to another.
+ */
 export class TranslateAction implements Action {
+    
+    /**
+     * An array of the selected component(s)
+     */
     protected objs: Component[];
+
+    /**
+     * An array of the starting position(s) of the selected component(s)
+     */
     protected initialPositions: Vector[];
+
+    /**
+     * An array of the final position(s) of the selected component(s)
+     */
     protected targetPositions: Vector[];
 
+    /**
+     * Creates a translation of component(s) from one position to another.
+     * Each component in objs list has corresponding initial position and target position in those
+     * respective lists.
+     * @param objs Initializes the array with the selected component(s)
```

Comment:
nit: add an empty line between the end of the description and the first tag like so:
```
     * respective lists.
     *
     * @param objs Initializes the array with the selected component(s)
```